bin/describe.ml

```diff
@@ -747,61 +753,96 @@ module Preprocess = struct
     let file_in_build_dir =
       if String.is_empty file then
         User_error.raise [ Pp.textf "no file is given" ]
-      else Path.of_string file |> in_build_dir |> Path.build
+      else Path.of_string file |> in_build_dir
     in
     let pp_file =
-      file_in_build_dir |> Path.map_extension ~f:(fun ext -> ".pp" ^ ext)
+      file_in_build_dir |> Path.Build.map_extension ~f:(fun ext -> ".pp" ^ ext)
     in
-    Build_system.file_exists pp_file >>= function
+    let* project = Source_tree.root () >>| Source_tree.Dir.project in
+    let* pp_file_exists = Build_system.file_exists (Path.build pp_file) in
+    match pp_file_exists with
     | true ->
-      let* () = Build_system.build_file pp_file in
-      let+ project = Source_tree.root () >>| Source_tree.Dir.project in
-      Ok (project, pp_file)
+      let pp_file = Path.build pp_file in
+      let+ () = Build_system.build_file pp_file in
+      Ok (`Pp_file (project, pp_file))
     | false -> (
-      Build_system.file_exists file_in_build_dir >>= function
+      let* file_exists =
+        Build_system.file_exists (Path.build file_in_build_dir)
```

Comment:
Instead of doing this check, can we do `find_lib_or_exe`? Then we would know if we need this .pp extension based on the preprocessing spec.